ahbottomnavigation/src/main/java/com/aurelhubert/ahbottomnavigation/AHBottomNavigation.java

```diff
@@ -152,7 +155,7 @@ protected Parcelable onSaveInstanceState() {
 		Bundle bundle = new Bundle();
 		bundle.putParcelable("superState", super.onSaveInstanceState());
 		bundle.putInt("current_item", currentItem);
-		bundle.putStringArray("notifications", notifications);
+        bundle.putParcelableArrayList("notifications", (ArrayList<Notification>) notifications);
```

Comment:
It's unsafe to cast your List to ArrayList. 
If you change Notification.generateEmptyList(..) realization it's possible to get ClassCastException
Use new ArrayList(notifications);